### ISSUE TITLE
Convention Proposal: Pushed 5's Ejection

### ISSUE THREAD

**Issue by pianoblook** [Zamiell/hanabi-conventions#232]
Now that a _Trash Push_ on junk signals a Discharge (in the same way as an _Unkown Trash Discharge_), there's an opportunity for an Ejection using similar logic. Just like those Discharges promise the focused card to be trash, this would promise the focused card as a 5 (just like 5CE). It would also complete the cycle of Play -> Finesse -> Ejection -> Discharge for trash pushes using the same logic as normal color clues.

Trash-Pushed 5's Ejection:
- Normally, if a player touches known trash on a teammate's chop position, it is a _Trash Push_ and calls for them to blind play the 'pushed' card.
- If the next player sees that the _Trash Pushed_ card would only be playable through one blind play, they should blind-play their _Finesse Position_ card (this is a _Trash Push Finesse_).
- If the pushed card is a 5 and would only be playable through **more than one blind play**, they should instead play their _Second Finesse Position* as a _Pushed 5's Ejection_.
- If the pushed card is trash, this instead signals a _Trash Push Discharge_. 

Example:
- All 1's are played.
- Alice clues 1 to Cathy's chop card, signaling a *Trash Push*
- Bob sees the *Pushed* card is a 5 which is far from playable.
- Bob blind plays his second finesse position, signaling to Cathy that her pushed card is actually a nonplayable 5. (And therefore 'chop moves' herself, technically?)

_Junk Ejection (w/ 5's)_ already exists, but this could be more useful in some situations, e.g. in variants with Brownish suits and/or in situations where there's value in getting other trash cards touched.

**Comment by Zamiell** [comment]
a link to a game where it was used (no hypotheticals), or a link to a game where it would have been optimal to use on a specific turn

**Comment by Dr-Kakashi** [comment]
This follows in line with what I've proposed here: https://github.com/Zamiell/hanabi-conventions/issues/186

**Comment by pianoblook** [comment]
Ah, I see this has indeed been proposed before (in Jeff's linked [doc](https://docs.google.com/document/d/1vxd8ecOIRnP5HtGZ3toyTsY-w0Yldir3D5cBN241M_U/edit?ts=5d66152e#heading=h.9zcbnfec21c7) too.) 

I'll keep my eyes out for examples! I can't find any screenshots of existing _Junk Ejection (w/ 5's)_ but I'd be surprised if this wouldn't be superior in some of them

**Comment by Dr-Kakashi** [comment]
It doesn't really matter who proposed it or who did it first....what really matters is that it follows the concepts of what we've established in the community.  

I personally feel this should be a thing because it tells players whether or not to keep the trash pushed card.

**Comment by Dr-Kakashi** [comment]
I was thinking about it some more....what if the 5 was blocked.....like... 

Green suit is finished. R2 has been played
Cathy's Hand state:
x - x - R1 - R5 - G1
Alice clue's green to Cathy pushing the R5
Bob understands that R5 is being pushed.
Bob understands that this is a trash push ejection and ejects
Cathy marks card to the left of trash as 345 depending on the board state at the time.

**Comment by iamwhoiamhahaha** [comment]
I believe this is worse than my original proposal lol. Let's just try to do some TPE and link the game to convince Zamiel to add it to the doc.

**Comment by Dr-Kakashi** [comment]
I'm misunderstanding something it's exactly what you've proposed isn't it? 

I've just looked at your doc.

**Comment by Zamiell** [comment]
still waiting on evidence

**Comment by pianoblook** [comment]
I think I agree that limiting _Push Ejections_ to only 5s is probably too restricting, especially with 5CE already being better.

Jeff's [proposal](https://github.com/Zamiell/hanabi-conventions/issues/257) is to allow using _any_  >1-away-from-playable, which seems fine but also perhaps too loose.

As a compromise between flexibility and added informational value, what if this kind of Pushed Ejection should promise a critical card, or, at the very least, that the 'pushed' card should be known _CM_'d?

As a counter-example, we restrict _Color Ejections_ to only 5s, not just any more-than-one-away card. Although this limits the number of times it would arise, it significantly increases the value of the clue. Thoughts?

**Comment by iamwhoiamhahaha** [comment]
I have proposed these long time ago: [Trash Push Ejection](https://docs.google.com/document/d/1vxd8ecOIRnP5HtGZ3toyTsY-w0Yldir3D5cBN241M_U/edit?ts=5d66152e#heading=h.qxcgfa4lsqji) and [Trash Push Discharge](https://docs.google.com/document/d/1vxd8ecOIRnP5HtGZ3toyTsY-w0Yldir3D5cBN241M_U/edit?ts=5d66152e#heading=h.2edw83yah7f1). The only thing we have to do right now is to post screenshots and convince Zamiel to make TPE official.

> Jeff's proposal is to allow using any >1-away-from-playable, which seems fine but also perhaps too loose.

I don't see why your proposal is more useful than mine. If Zamiel rejected my proposal 8 months ago, then he'll never approve your proposal, since the possibility to do a TPE by your definition is pretty lower than mine.

> As a compromise between flexibility and added informational value, what if this kind of Pushed Ejection should promise a critical card, or, at the very least, that the 'pushed' card should be known CM'd?

It does *Chop Move* in my original proposal no matter if it is critical. See [Trash Push Ejection Chop Move](https://docs.google.com/document/d/1vxd8ecOIRnP5HtGZ3toyTsY-w0Yldir3D5cBN241M_U/edit?ts=5d66152e#heading=h.7p6v9h5kihsb).

**Comment by Zamiell** [comment]
i agree that the Jeff's formulation of TPE seems more useful than the version where it can only be a 5.

regardless, Jeff's formulation of TPE it is still:

1) a convention that would be rarely-used and add to the bloat
2) a convention that would be easy-to-forget
3) a convention that is just a 1-for-1
3) a convention that would only apply in edge-case scenarios where all "normal" clues are blocked (e.g. would probably only happen in some fucked up brown variant)

(note that **even if** you have the weird TPE chop move thing, TPE would still only be a 1 for 1, because chop moving cards doesn't count when calculating efficiency, not that I am even a fan of the TPE chop move in the first place)

so these are the reasons why TPE was rejected 8 months ago. has anything changed since then? probably not much. regardless of whether we are using piano's version or jeff's version, we need substantial evidence that it will be a useful move.

i'll leave this issue open for a while longer.

**Comment by pianoblook** [comment]
Just saw a pretty funky but elegant opportunity for this in a replay today: https://hanabi.live/replay/207576/18

<img width="1213" alt="Screen Shot 2020-06-15 at 10 39 05 AM" src="https://user-images.githubusercontent.com/32303453/84670691-710aae00-aef4-11ea-8fbc-2aecdf2de891.png">

-asaelr hints '1' to Zakisan as a _Trash Push_
-sephiroth must _Eject_ to demonstrate the 5
-Now it's Zakisan's turn. his y2 is fully known, and he knows sephiroth is about to misplay g2 as r2.
-By Ejecting a duplicate of a known playable, it's now been forced into a reverse trash bluff. Zakisan blindplays slot 1 for the matching r2

**Comment by Zamiell** [comment]
the screenshot isn't a valid demonstration of the convention in question

going to close this now, since it seems like the discussion is dying out
